topic0/src/abstract_factory/Demo.java

```diff
@@ -0,0 +1,22 @@
+package abstract_factory;
+
+public class Demo {
+	public static void main(String[] args) {
+
+		/*
+		 * Each type of database factory contains two types of databases: Accounts and
+		 * Clients. When a dataabase factory is ordered, we specify which type of
+		 * database we wish to connect to. Each string test represents a type of
+		 * database from a specified factory.
+		 */
+		DatabaseFactory mysqlConnection = ConnectionMaker.establishConnection("mysql");
+		DatabaseFactory mongodbConnection = ConnectionMaker.establishConnection("mongodb");
+
+		String mysqlAccounts = mysqlConnection.getMysql("accounts").connectTo();
```

Comment:
As I said the pattern is not working properly in your code, but don't worry what we wanted is you to play with these kind of code. If you have some time take another look at it later on.